src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/RoundaboutValenceCheck.java

```diff
@@ -0,0 +1,187 @@
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Edge edge = (Edge) object;
+        final Map<Long, Edge> roundaboutEdges = new HashMap<>();
+
+        // Get all edges in the roundabout
+        getAllRoundaboutEdges(edge, roundaboutEdges);
+        final int totalRoundaboutValence;
+        final int roundaboutEdgeCount = roundaboutEdges.size();
+        final Set<Edge> connectedEdges = new HashSet<>();
+
+        final Iterator iterator = roundaboutEdges.entrySet().iterator();
+
+        if (!iterator.hasNext())
+        {
+            return Optional.empty();
+        }
+
+        while (iterator.hasNext())
+        {
+            final Map.Entry pair = (Map.Entry) iterator.next();
+            final Edge roundaboutEdge = (Edge) pair.getValue();
+
+            // Adds only car navigable master edges that are connected to the roundabout toward
+            // the connectedEdges set
+            connectedEdges.addAll(roundaboutEdge.connectedEdges().stream().map(Edge::getMasterEdge)
+                    .filter(e -> HighwayTag.isCarNavigableHighway(e) || JunctionTag.isRoundabout(e))
+                    .collect(Collectors.toSet()));
+        }
+
+        totalRoundaboutValence = connectedEdges.size() - roundaboutEdgeCount;
+
+        // If the totalRoundaboutValence is less than the minimum configured number of connections
+        // or greater than or equal to the maximum configured number of connections
+        if (totalRoundaboutValence < this.minimumValence
+                || totalRoundaboutValence >= this.maximumValence)
+        {
+            this.markAsFlagged(object.getIdentifier());
+
+            // If the roundabout valence is 1, this should be labelled as a turning loop instead
+            if (totalRoundaboutValence == 1)
+            {
+                return Optional.of(this.createFlag(connectedEdges,
+                        this.getLocalizedInstruction(1, edge.getOsmIdentifier())));
+            }
+            // Otherwise, we want to flag and given information about identifier and valence
+            return Optional.of(this.createFlag(connectedEdges,
+                    this.getLocalizedInstruction(0, edge.getOsmIdentifier(), totalRoundaboutValence,
+                            roundaboutEdgeCount, connectedEdges.size())));
+        }
+        // If the totalRoundaboutValence is not unusual, we don't flag the object
+        else
+        {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * This method gets all edges in a roundabout given one edge in that roundabout
+     * 
+     * @param edge
+     *            An Edge object known to be a roundabout edge
+     * @param roundaboutEdges
+     *            A map that contains key, value pairs of Edge identifiers and all associated Edge
+     *            data
+     */
+    private void getAllRoundaboutEdges(final Edge edge, final Map<Long, Edge> roundaboutEdges)
```

Comment:
For the reasons above can you refactor this to ```private Set<Edge> getAllRoundaboutEdges(final Edge edge)```

Reply:
Refactored. Let me know if this wasn't what you were thinking!